packages/dashboard-core-plugins/src/panels/MarkdownNotebookPanel.jsx

```diff
@@ -0,0 +1,392 @@
+    } else {
+      this.tabTitleElement.classList.remove('is-preview');
+    }
+  }
+
+  registerFileMetadata(fileMetadata, isPreview) {
+    const { glEventHub, id: tabId } = this.props;
+    glEventHub.emit(
+      NotebookEvent.REGISTER_FILE,
+      tabId,
+      fileMetadata,
+      isPreview
+    );
+  }
+
+  setEditorPosition(clickPositionY) {
+    const { container: markdownEditorContainer } = this.markdownEditor;
+    if (this.editor && markdownEditorContainer) {
+      const contentTop = markdownEditorContainer.getBoundingClientRect().top;
+      const contentScrollTop = markdownEditorContainer.scrollTop;
+      const contentScrollHeight = markdownEditorContainer.scrollHeight;
+      const totalLines = this.editor.getModel().getLineCount();
+
+      let lineToFocus = Math.round(
+        ((contentScrollTop + clickPositionY - contentTop) /
+          contentScrollHeight) *
+          totalLines
+      );
+      if (lineToFocus > totalLines) {
+        lineToFocus = totalLines;
+      }
+
+      this.editor.revealLine(lineToFocus);
+      this.editor.setPosition({
+        lineNumber: lineToFocus,
+        column: 1,
+      });
+      this.editor.focus();
+    }
+  }
+
+  handleContainerDoubleClick(event) {
+    const { isEditing } = this.state;
+    const dbClickPositionY = event.clientY;
+
+    if (!isEditing) {
+      this.setState({ isEditing: true }, () => {
+        this.setEditorPosition(dbClickPositionY);
+      });
+    }
+  }
+
+  handleEditorInitialized(editor) {
+    log.debug('Markdown Editor Initialized...');
+    this.editor = editor;
+  }
+
+  handleEditorBlur(event) {
+    log.debug(`markdown content changed, saving...`);
+    const { isEditing } = this.state;
+
+    // if not in edit mode, or in edit mode but blur went to an internal monaco field (like search)
+    if (
+      !isEditing ||
+      this.markdownEditor?.container?.contains(event.relatedTarget)
+    ) {
+      return;
+    }
+
+    const content = this.editor.getValue();
+
+    this.setState({
+      content,
+      isEditing: false,
+    });
+  }
+
+  handleEditorResize() {
+    const { isEditing } = this.state;
+    if (isEditing && this.editor) {
+      this.editor.layout();
+    }
+  }
+
+  /**
+   * @param {MouseEvent} event The click event from clicking on the link
+   */
+  handleLinkClick(event) {
+    const { notebooksUrl, session, sessionLanguage } = this.props;
+    const { href } = event.target;
+    if (!href || !href.startsWith(notebooksUrl)) {
+      return;
+    }
+
+    event.stopPropagation();
+    event.preventDefault();
+
+    const notebookPath = `/${href.substring(notebooksUrl.length)}`.replace(
+      /%20/g,
+      ' '
+    );
+    log.debug('Notebook link clicked, opening', notebookPath);
+
+    const { glEventHub } = this.props;
+    const notebookSettings = {
+      value: null,
+      language: sessionLanguage,
+    };
+    const fileMetadata = { id: notebookPath, itemName: notebookPath };
+    glEventHub.emit(
```

Comment:
I noticed the notebooks sometimes don't reuse the tabs that are already open. Could be related to the `props.id` issue above.

Reply:
I don't think this is an issue anymore